synapse/storage/databases/main/end_to_end_keys.py

```diff
@@ -221,41 +250,32 @@ def _get_e2e_device_keys_and_signatures_txn(
             for user_id, device_id in deleted_devices:
                 result.setdefault(user_id, {})[device_id] = None
 
-        # get signatures on the device
-        signature_sql = ("SELECT *  FROM e2e_cross_signing_signatures WHERE %s") % (
-            " OR ".join("(" + q + ")" for q in signature_query_clauses)
-        )
-
-        txn.execute(signature_sql, signature_query_params)
-        rows = self.db_pool.cursor_to_dict(txn)
-
-        # add each cross-signing signature to the correct device in the result dict.
-        for row in rows:
-            signing_user_id = row["user_id"]
-            signing_key_id = row["key_id"]
-            target_user_id = row["target_user_id"]
-            target_device_id = row["target_device_id"]
-            signature = row["signature"]
-
-            target_user_result = result.get(target_user_id)
-            if not target_user_result:
-                continue
+        return result
 
-            target_device_result = target_user_result.get(target_device_id)
-            if not target_device_result:
-                # note that target_device_result will be None for deleted devices.
-                continue
+    def _get_e2e_cross_signing_signatures_for_devices_txn(
+        self, txn: Cursor, device_query: Iterable[Tuple[str, str]]
+    ) -> List[Dict]:
+        """Get cross-signing signatures for a given list of devices
 
-            target_device_signatures = target_device_result.signatures
-            if target_device_signatures is None:
-                target_device_signatures = target_device_result.signatures = {}
+        Returns signatures made by the owner of the devices. Each entry in the result
+        is a dict containing the fields from the database ('user_id', 'key_id',
+        'target_user_id', 'target_device_id', 'signature').
```

Comment:
Not sure if it really matters but don't we know that `user_id == target_user_id` here? Seems silly to return them separately (as we're just pumping more bytes over the database connection). But maybe this is trying to micro-optimize.

Reply:
well you asked for it; it now avoids pulling the spurious column :)